engines/native/sandbox.go

```diff
@@ -35,6 +38,13 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		return nil, fmt.Errorf("Failed to temporary folder, error: %s", err)
 	}
 
+	if b.payload.Context != "" {
+		if err = fetchContext(b.payload.Context, homeFolder.Path()); err != nil {
```

Comment:
Add a TODO to start downloading at creation of the sandbox builder, and wait for download to finish here.
Add another TODO to make that abortable.